tests/suites/test_suite_ecdh.function

```diff
@@ -50,6 +50,8 @@ void ecdh_invalid_param( )
     mbedtls_ecp_keypair kp;
     int invalid_side = 42;
 
+    mbedtls_ecp_keypair_init( &kp );
+
     TEST_EQUAL( MBEDTLS_ERR_ECP_BAD_INPUT_DATA,
                             mbedtls_ecdh_get_params( &ctx, &kp,
                                                      invalid_side ) );
```

Comment:
As David says, this isn't really a good test. Ideally it should start with a `mbedtls_ecdh_get_params()` call that succeeds, so we know all the parameters are valid, and then it should vary each one in turn to be invalid, and check for the correct return value. This could just be working because the `side` parameter isn't `MBEDTLS_ECDH_OURS` or `MBEDTLS_ECDH_THEIRS`, or because the key is uninitialised (here gcc has definitely caught an issue, as the keypair is an input parameter).